tests/tests-pending/rpc_pending_test.go

```diff
@@ -0,0 +1,326 @@
+	"math/big"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+
+	rpctypes "github.com/cosmos/ethermint/rpc/types"
+	util "github.com/cosmos/ethermint/tests"
+)
+
+const (
+	addrA         = "0xc94770007dda54cF92009BFF0dE90c06F603a09f"
+	addrAStoreKey = 0
+)
+
+var (
+	MODE = os.Getenv("MODE")
+	from = []byte{}
+)
+
+type Request struct {
+	Version string      `json:"jsonrpc"`
+	Method  string      `json:"method"`
+	Params  interface{} `json:"params"`
+	ID      int         `json:"id"`
+}
+
+type RPCError struct {
+	Code    int         `json:"code"`
+	Message string      `json:"message"`
+	Data    interface{} `json:"data,omitempty"`
+}
+
+type Response struct {
+	Error  *RPCError       `json:"error"`
+	ID     int             `json:"id"`
+	Result json.RawMessage `json:"result,omitempty"`
+}
+
+func TestMain(m *testing.M) {
+	if MODE != "pending" {
+		_, _ = fmt.Fprintln(os.Stdout, "Skipping pending RPC test")
+		return
+	}
+
+	var err error
+	from, err = util.GetAddress()
+	if err != nil {
+		fmt.Printf("failed to get account: %s\n", err)
+		os.Exit(1)
+	}
+
+	// Start all tests
+	code := m.Run()
+	os.Exit(code)
+}
+
+func TestEth_Pending_GetBalance(t *testing.T) {
+	var res hexutil.Big
+	rpcRes := util.Call(t, "eth_getBalance", []string{addrA, "latest"})
+	err := res.UnmarshalJSON(rpcRes.Result)
+	require.NoError(t, err)
+	preTxLatestBalance := res.ToInt()
+
+	rpcRes = util.Call(t, "eth_getBalance", []string{addrA, "pending"})
+	err = res.UnmarshalJSON(rpcRes.Result)
+	require.NoError(t, err)
+	preTxPendingBalance := res.ToInt()
+
+	t.Logf("Got pending balance %s for %s pre tx\n", preTxPendingBalance, addrA)
+	t.Logf("Got latest balance %s for %s pre tx\n", preTxLatestBalance, addrA)
+
+	param := make([]map[string]string, 1)
+	param[0] = make(map[string]string)
+	param[0]["from"] = "0x" + fmt.Sprintf("%x", from)
+	param[0]["to"] = addrA
+	param[0]["value"] = "0xA"
+	param[0]["gasLimit"] = "0x5208"
+	param[0]["gasPrice"] = "0x1"
+
+	_ = util.Call(t, "eth_sendTransaction", param)
+
+	rpcRes = util.Call(t, "eth_getBalance", []string{addrA, "pending"})
+	err = res.UnmarshalJSON(rpcRes.Result)
+	require.NoError(t, err)
+	postTxPendingBalance := res.ToInt()
+	t.Logf("Got pending balance %s for %s post tx\n", postTxPendingBalance, addrA)
+
+	require.Equal(t, preTxPendingBalance.Add(preTxPendingBalance, big.NewInt(10)), postTxPendingBalance)
+
+	rpcRes = util.Call(t, "eth_getBalance", []string{addrA, "latest"})
+	err = res.UnmarshalJSON(rpcRes.Result)
+	require.NoError(t, err)
+	postTxLatestBalance := res.ToInt()
+	t.Logf("Got latest balance %s for %s post tx\n", postTxLatestBalance, addrA)
+
+	require.Equal(t, preTxLatestBalance, postTxLatestBalance)
+	// 1 if x > y; where x is postTxPendingBalance, y is preTxPendingBalance
+	if postTxLatestBalance.Cmp(preTxPendingBalance) != 0 {
+		t.Errorf("expected balance: %d, got: %s", 0, res.String())
+	}
```

Comment:
this check is not necessary because it's already checked on the line above. Unless you want to compare against `0` (which is what the log states?)

Reply:
absolutely correct. removing